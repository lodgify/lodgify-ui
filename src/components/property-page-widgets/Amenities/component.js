import React from 'react';
import PropTypes from 'prop-types';

import { VIEW_MORE } from 'utils/default-strings';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { Heading } from 'typography/Heading';

import { getDefaultItems } from './utils/getDefaultItems';
import { hasExtraItems } from './utils/hasExtraItems';
import { getCategoryMarkup } from './utils/getCategoryMarkup';
import { getExtraItemsMarkup } from './utils/getExtraItemsMarkup';
import { getExtraItems } from './utils/getExtraItems';

/**
 * The standard widget for displaying the amenities of a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  amenities,
  hasExtraItemsInModal,
  headingText,
  isStacked,
  modalTriggerText,
}) => (
  <Grid className="is-amenities" columns={isStacked ? 1 : 3}>
    <GridRow>
      {headingText && (
        <GridColumn width={12}>
          <Heading>{headingText}</Heading>
        </GridColumn>
      )}
      {getDefaultItems(amenities, isStacked).map((amenity, index) =>
        getCategoryMarkup(amenity, index, isStacked)
      )}
      {hasExtraItems(amenities, isStacked) &&
        getExtraItemsMarkup(
          hasExtraItemsInModal,
          modalTriggerText,
          getExtraItems(amenities, isStacked)
        )}
    </GridRow>
  </Grid>
);

Component.displayName = 'Amenities';

Component.defaultProps = {
  hasExtraItemsInModal: false,
  headingText: null,
  isStacked: false,
  modalTriggerText: VIEW_MORE,
};

Component.propTypes = {
  /** The amenity categories. */
  amenities: PropTypes.arrayOf(
    PropTypes.shape({
      /** The name of the icon to display for the category.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** The list of amenities */
      items: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
      /** The category name */
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
  /** Are the extra items displayed in a modal. */
  hasExtraItemsInModal: PropTypes.bool,
  /** The text to display as a heading at the top of the amenities. */
  headingText: PropTypes.string,
  /** Are the amenities displayed stacked on top of one another */
  isStacked: PropTypes.bool,
  /** The text for the modal trigger */
  modalTriggerText: PropTypes.string,
};
