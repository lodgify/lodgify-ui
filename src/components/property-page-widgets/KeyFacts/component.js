import React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'semantic-ui-react';

import { KEY_FACTS } from 'utils/default-strings';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Heading } from 'typography/Heading';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { IconCard } from 'elements/IconCard';

/**
 * The standard widget for displaying key facts about a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({ keyFacts, headingText }) => (
  <Grid>
    <GridColumn width={12}>
      <Heading>{headingText}</Heading>
    </GridColumn>
    <GridColumn width={12}>
      <Label.Group>
        {keyFacts.map(({ iconName, isDisabled, label }, index) => (
          <IconCard
            isDisabled={isDisabled}
            isFilled
            key={buildKeyFromStrings(label, index)}
            label={label}
            name={iconName}
          />
        ))}
      </Label.Group>
    </GridColumn>
  </Grid>
);

Component.displayName = 'KeyFacts';

Component.defaultProps = {
  headingText: KEY_FACTS,
};

Component.propTypes = {
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The key facts to display as icon cards. */
  keyFacts: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * The name of the icon to display.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** Is the key fact disabled. */
      isDisabled: PropTypes.bool,
      /** A visible label to display for the key fact. */
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
};
