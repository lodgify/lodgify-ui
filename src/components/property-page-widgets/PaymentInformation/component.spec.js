import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as PaymentInformation } from './component';

const paymentScheduleText = `50% due at time of booking. Remaining balance: Due later.`;
const cancellationPolicyText = `All paid prepayments are non-refundable.`;
const cleaningCharge = `25$ (USD)`;
const taxesText = `1%`;
const taxesDescriptionText = `of total booking value`;
const damageDepositText = `A refundable damage deposit of 200.00 â‚¬ (EUR) is due.`;
const notesText = `
  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
`;
const extraNotesText = `
  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.

  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
`;

const props = {};

const getPaymentInformation = extraProps =>
  mount(<PaymentInformation {...props} {...extraProps} />);

describe('<PaymentInformation />', () => {
  it('should have the right structure', () => {
    const wrapper = getPaymentInformation();

    expect(wrapper).toMatchSnapshot();
  });

  describe('if `props.paymentScheduleText` is passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({ paymentScheduleText });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.cancellationPolicyText` is passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({ cancellationPolicyText });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.cleaningCharge` is passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({ cleaningCharge });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.taxesText` and `props.taxesDescriptionText` are passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({
        taxesDescriptionText,
        taxesText,
      });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.damageDepositText` is passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({
        damageDepositText,
      });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.notesText` is passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({
        notesText,
      });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.extraNotesText` is passed', () => {
    it('should have the right structure', () => {
      const wrapper = getPaymentInformation({
        extraNotesText,
      });

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should have `displayName` `PaymentInformation`', () => {
    expectComponentToHaveDisplayName(PaymentInformation, 'PaymentInformation');
  });
});
