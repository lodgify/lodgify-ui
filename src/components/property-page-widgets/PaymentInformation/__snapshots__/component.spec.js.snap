// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentInformation /> if \`props.cancellationPolicyText\` is passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText="All paid prepayments are non-refundable."
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={6}
              >
                <GridColumn
                  verticalAlign="top"
                  width={6}
                >
                  <div
                    className="top aligned six wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="small"
                      textAlign={null}
                    >
                      <Header
                        as="h4"
                        inverted={false}
                        textAlign={null}
                      >
                        <h4
                          className="ui header"
                        >
                          Cancellation Policy
                        </h4>
                      </Header>
                    </Heading>
                    <Paragraph
                      size="medium"
                      weight={null}
                    >
                      <p
                        className=""
                      >
                        All paid prepayments are non-refundable.
                      </p>
                    </Paragraph>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> if \`props.cleaningCharge\` is passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge="25$ (USD)"
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={6}
              >
                <GridColumn
                  verticalAlign="top"
                  width={6}
                >
                  <div
                    className="top aligned six wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="small"
                      textAlign={null}
                    >
                      <Header
                        as="h4"
                        inverted={false}
                        textAlign={null}
                      >
                        <h4
                          className="ui header"
                        >
                          Cleaning Charge
                        </h4>
                      </Header>
                    </Heading>
                    <Statistic
                      horizontal={true}
                      size="mini"
                      text={true}
                      value="25$ (USD)"
                    >
                      <div
                        className="ui mini horizontal statistic"
                      >
                        <StatisticValue
                          content="25$ (USD)"
                          key="25$ (USD)"
                          text={true}
                        >
                          <div
                            className="text value"
                          >
                            25$ (USD)
                          </div>
                        </StatisticValue>
                      </div>
                    </Statistic>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> if \`props.damageDepositText\` is passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText="A refundable damage deposit of 200.00 â‚¬ (EUR) is due."
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="small"
                      textAlign={null}
                    >
                      <Header
                        as="h4"
                        inverted={false}
                        textAlign={null}
                      >
                        <h4
                          className="ui header"
                        >
                          Damage Deposit
                        </h4>
                      </Header>
                    </Heading>
                    <Paragraph
                      size="medium"
                      weight={null}
                    >
                      <p
                        className=""
                      >
                        A refundable damage deposit of 200.00 â‚¬ (EUR) is due.
                      </p>
                    </Paragraph>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> if \`props.extraNotesText\` is passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText="
  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.

  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
"
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Modal
                      isFullscreen={false}
                      size="tiny"
                      trigger={
                        <Link
                          href={null}
                          isPositionedRight={false}
                          onClick={[Function]}
                          willOpenInNewTab={false}
                        >
                          View more
                        </Link>
                      }
                    >
                      <Modal
                        closeIcon={
                          <Icon
                            color={null}
                            hasBorder={false}
                            isCircular={false}
                            isColorInverted={false}
                            isDisabled={false}
                            isLabelLeft={false}
                            labelText={null}
                            labelWeight={null}
                            name="close"
                            path={null}
                            size={null}
                          />
                        }
                        closeOnDimmerClick={true}
                        closeOnDocumentClick={false}
                        content={
                          Array [
                            <Paragraph
                              size="medium"
                              weight={null}
                            >
                              Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.
                            </Paragraph>,
                            <Paragraph
                              size="medium"
                              weight={null}
                            >
                              Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
                            </Paragraph>,
                            <Paragraph
                              size="medium"
                              weight={null}
                            >
                              Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.
                            </Paragraph>,
                            <Paragraph
                              size="medium"
                              weight={null}
                            >
                              Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
                            </Paragraph>,
                          ]
                        }
                        dimmer="inverted"
                        eventPool="Modal"
                        size="tiny"
                        trigger={
                          <Link
                            href={null}
                            isPositionedRight={false}
                            onClick={[Function]}
                            willOpenInNewTab={false}
                          >
                            View more
                          </Link>
                        }
                      >
                        <Portal
                          className="ui inverted page modals dimmer transition visible active"
                          closeOnDocumentClick={false}
                          closeOnEscape={true}
                          closeOnRootNodeClick={true}
                          eventPool="Modal"
                          mountNode={<body />}
                          onClose={[Function]}
                          onMount={[Function]}
                          onOpen={[Function]}
                          onUnmount={[Function]}
                          openOnTriggerClick={true}
                          trigger={
                            <Link
                              href={null}
                              isPositionedRight={false}
                              onClick={[Function]}
                              willOpenInNewTab={false}
                            >
                              View more
                            </Link>
                          }
                        >
                          <Ref
                            innerRef={[Function]}
                          >
                            <Link
                              href={null}
                              isPositionedRight={false}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              willOpenInNewTab={false}
                            >
                              <Button
                                as={null}
                                basic={true}
                                floated="left"
                                href={null}
                                onClick={[Function]}
                                target="_self"
                              >
                                <button
                                  className="ui basic left floated button"
                                  href={null}
                                  onClick={[Function]}
                                  role="button"
                                  target="_self"
                                >
                                  View more
                                </button>
                              </Button>
                            </Link>
                          </Ref>
                        </Portal>
                      </Modal>
                    </Modal>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> if \`props.notesText\` is passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText="
  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
"
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="small"
                      textAlign={null}
                    >
                      <Header
                        as="h4"
                        inverted={false}
                        textAlign={null}
                      >
                        <h4
                          className="ui header"
                        >
                          Notes
                        </h4>
                      </Header>
                    </Heading>
                    <Paragraph
                      size="medium"
                      weight={null}
                    >
                      <p
                        className=""
                      >
                        
  Livingstone is a modern website template with clean and straight lines. Its special feature is a wide horizontal header photo slideshow in which logo and header navigation nicely blend in.

  Maecenas et efficitur diam. Etiam non ante urna. Donec imperdiet cursus lectus, luctus vestibulum urna aliquet vel. Donec non vehicula est. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.

                      </p>
                    </Paragraph>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> if \`props.paymentScheduleText\` is passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText="50% due at time of booking. Remaining balance: Due later."
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={6}
              >
                <GridColumn
                  verticalAlign="top"
                  width={6}
                >
                  <div
                    className="top aligned six wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="small"
                      textAlign={null}
                    >
                      <Header
                        as="h4"
                        inverted={false}
                        textAlign={null}
                      >
                        <h4
                          className="ui header"
                        >
                          Payment Schedule
                        </h4>
                      </Header>
                    </Heading>
                    <Paragraph
                      size="medium"
                      weight={null}
                    >
                      <p
                        className=""
                      >
                        50% due at time of booking. Remaining balance: Due later.
                      </p>
                    </Paragraph>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> if \`props.taxesText\` and \`props.taxesDescriptionText\` are passed should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText="of total booking value"
  taxesHeadingText="Taxes"
  taxesText="1%"
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={6}
              >
                <GridColumn
                  verticalAlign="top"
                  width={6}
                >
                  <div
                    className="top aligned six wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="small"
                      textAlign={null}
                    >
                      <Header
                        as="h4"
                        inverted={false}
                        textAlign={null}
                      >
                        <h4
                          className="ui header"
                        >
                          Taxes
                        </h4>
                      </Header>
                    </Heading>
                    <Statistic
                      horizontal={true}
                      label="of total booking value"
                      size="tiny"
                      text={true}
                      value="1%"
                    >
                      <div
                        className="ui tiny horizontal statistic"
                      >
                        <StatisticValue
                          content="1%"
                          key="1%"
                          text={true}
                        >
                          <div
                            className="text value"
                          >
                            1%
                          </div>
                        </StatisticValue>
                        <StatisticLabel
                          content="of total booking value"
                          key="of total booking value"
                        >
                          <div
                            className="label"
                          >
                            of total booking value
                          </div>
                        </StatisticLabel>
                      </div>
                    </Statistic>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;

exports[`<PaymentInformation /> should have the right structure 1`] = `
<PaymentInformation
  cancellationPolicyHeadingText="Cancellation Policy"
  cancellationPolicyText={null}
  cleaningCharge={null}
  cleaningChargeHeadingText="Cleaning Charge"
  damageDepositHeadingText="Damage Deposit"
  damageDepositText={null}
  extraNotesText={null}
  headingText="Payment Information"
  modalTriggerText="View more"
  notesHeadingText="Notes"
  notesText={null}
  paymentScheduleHeadingText="Payment Schedule"
  paymentScheduleText={null}
  taxesDescriptionText={null}
  taxesHeadingText="Taxes"
  taxesText={null}
>
  <Grid
    areColumnsCentered={false}
    stackable={true}
  >
    <Grid
      centered={false}
      stackable={true}
    >
      <div
        className="ui stackable grid"
      >
        <GridRow
          horizontalAlignContent="left"
        >
          <GridRow
            textAlign="left"
          >
            <div
              className="left aligned row"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <GridColumn
                  verticalAlign="top"
                  width={12}
                >
                  <div
                    className="top aligned twelve wide column"
                  >
                    <Heading
                      isColorInverted={false}
                      size="medium"
                      textAlign={null}
                    >
                      <Header
                        as="h3"
                        inverted={false}
                        textAlign={null}
                      >
                        <h3
                          className="ui header"
                        >
                          Payment Information
                        </h3>
                      </Header>
                    </Heading>
                  </div>
                </GridColumn>
              </GridColumn>
            </div>
          </GridRow>
        </GridRow>
      </div>
    </Grid>
  </Grid>
</PaymentInformation>
`;
